datumaro/datumaro/plugins/image_dir.py

```diff
@@ -79,11 +73,5 @@ def __call__(self, extractor, save_dir):
 
         for item in extractor:
             if item.has_image and item.image.has_data:
-                filename = item.image.filename
-                if filename:
-                    filename = osp.splitext(filename)[0]
-                else:
-                    filename = item.id
-                filename += '.jpg'
-                save_image(osp.join(save_dir, filename), item.image.data,
-                    create_dir=True)
\ No newline at end of file
+                save_image(osp.join(save_dir, item.id + '.jpg'),
```

Comment:
What if the original images were PNG or any other format? Don't you have all these issues just because item.id doesn't the the original extension?

Reply:
>  all these issues

Which ones? 

- Relative paths? They have no relation to extensions and are not used in any dataset format.
- Appending an extension in converters? Maybe, but the extension, as well as fill image path, is available in `item.image.path`. And an option to copy original images files instead of recoding them is a matter of further PRs, as well as keeping the original extension.